### PR TITLE
Add datagram support.

### PR DESCRIPTION
**Pull request by jianjunz**

### CODE REVIEW

File: web_transport/sdk/api/owt/quic/web_transport_definitions.h [hunk 1]
```diff
@@ -31,11 +31,36 @@ struct OWT_EXPORT ConnectionStats {
...
   const char* fingerprint;
 };
 
+// Status of message being sent.
+// 1:1 mapping to MessageStatus in
+// net/third_party/quiche/src/quic/core/quic_types.h except kUnavailable.
+enum class MessageStatus {
+  // Message status is not available. When C++17 std::optional is enabled, this
+  // value will be removed.
+  kUnavailable,
```
Comment (taste1981) [inline web_transport/sdk/api/owt/quic/web_transport_definitions.h:46]: please kUnavaiable to end of the enum.

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc [hunk 2]
```diff
@@ -250,5 +258,38 @@ void WebTransportOwtClientImpl::FireEvent(
   }
 }
 
+MessageStatus WebTransportOwtClientImpl::SendOrQueueDatagram(uint8_t* data,
+                                                             size_t length) {
+  auto* allocator = client_->quic_session()
+                        ->connection()
```
Comment (taste1981) [inline web_transport/sdk/impl/web_transport_owt_client_impl.cc:264]: perform check by:
client_->quic_session()->connection()->connected() first before you go through the sending process.

File: web_transport/sdk/impl/web_transport_server_session.cc [hunk 3]
```diff
@@ -89,6 +90,35 @@ WebTransportServerSession::CreateBidirectionalStream() {
   return result;
 }
 
+MessageStatus WebTransportServerSession::SendOrQueueDatagram(uint8_t* data,
+                                                             size_t length) {
+  auto* allocator =
```
Comment (taste1981) [inline web_transport/sdk/impl/web_transport_server_session.cc:95]: ditto.

File: web_transport/sdk/api/owt/quic/web_transport_definitions.h (hunk 1 above)
Comment (jianjunz) [reply web_transport/sdk/api/owt/quic/web_transport_definitions.h]: Done.

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc (hunk 2 above)
Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_owt_client_impl.cc:267]: Added DCHECK.

File: web_transport/sdk/impl/web_transport_server_session.cc (hunk 3 above)
Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_server_session.cc:97]: Added DCHECK.

**Review by taste1981 — APPROVED** [review]

**Pull request merged**